Lib/Src/CThermistor.cpp

```diff
@@ -0,0 +1,54 @@
+/**
+ * @file CThermistor.cpp
+ */
+
+/*
+ *  Created on: Feb 15, 2022
+ *      Author: niuslar
+ */
+
+#include "CThermistor.h"
+#include "math.h"
+
+#define OUT_OF_RANGE_VALUE (999.9)
+
+CUartCom uart_temp(&huart2);
+CLog log_temp(&uart_temp, "Temp_sensor");
+
+/**
+ * @brief Constructor
+ */
+CThermistor::CThermistor() {}
+
+/**
+ * @brief replaces the default voltage range
+ * @param min_voltage Minimum voltage limit
+ * @param max_voltage Max voltage limit
+ */
+void CThermistor::setLimits(const float min_voltage, const float max_voltage)
+{
+    m_min_volt_limit = min_voltage;
+    m_max_volt_limit = max_voltage;
+}
+
+float CThermistor::getTemperature(float voltage)
```

Comment:
You can declare this method `const`. This way it can be optimised further by the compiler. Have a look what that means. Can also look at what `pure` method does.